pkg/storage/tsdb/config.go

```diff
@@ -146,6 +148,9 @@ func (cfg *Config) RegisterFlags(f *flag.FlagSet) {
 	f.BoolVar(&cfg.WALCompressionEnabled, "experimental.tsdb.wal-compression-enabled", false, "True to enable TSDB WAL compression.")
 	f.BoolVar(&cfg.StoreGatewayEnabled, "experimental.tsdb.store-gateway-enabled", false, "True if the Cortex cluster is running the store-gateway service and the querier should query the bucket store via the store-gateway.")
 	f.BoolVar(&cfg.FlushBlocksOnShutdown, "experimental.tsdb.flush-blocks-on-shutdown", false, "If true, and transfer of blocks on shutdown fails or is disabled, incomplete blocks are flushed to storage instead. If false, incomplete blocks will be reused after restart, and uploaded when finished.")
+
+	f.StringVar(&cfg.BackfillDir, "experimental.tsdb.backfill-dir", "backfill_tsdb", "Local directory to store backfill TSDBs in the ingesters.")
+	f.DurationVar(&cfg.BackfillLimit, "experimental.tsdb.backfill-limit", 6*time.Hour, "")
```

Comment:
Few comments:
1. I wouldn't call it "limit" but "max age" (so `backfill_max_age` and `.backfill-max-age`). Limit can be confused with our limits (like max series per user/metric).
2. Please add the flag description. I would say something like "Maximum accepted sample age by backfilling. 0 disables it."
3. I would support in the code a 0 value to disable it. Also, I would set a default to 0.

Reply:
> I would support in the code a 0 value to disable it. Also, I would set a default to 0.

That was the plan going ahead. I will make these changes.